universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramSummaryView.java

```diff
@@ -152,9 +158,18 @@ private ContainerTag renderQuestionSummary(
 
     ContainerTag answerContent;
     if (data.fileKey().isPresent()) {
-      String fileLink =
-          controllers.routes.FileController.show(applicantId, data.fileKey().get()).url();
-      answerContent = a().withHref(fileLink).withClasses(Styles.W_2_3);
+      try {
+        String encodedFileKey = URLEncoder.encode(data.fileKey().get(), StandardCharsets.UTF_8);
+        String fileLink = controllers.routes.FileController.show(applicantId, encodedFileKey).url();
+        answerContent = a().withHref(fileLink).withClasses(Styles.W_2_3);
+      } catch (Exception e) {
```

Comment:
Can we catch a more specific exception here?

Reply:
My IDE kept complaining that there's no exception thrown, so I took another look at the documentation: https://docs.oracle.com/javase/7/docs/api/java/net/URLEncoder.html. The exception is only thrown from the function if the encoding is not supported, but we're using a standard encoding, UTF 8, which is "guaranteed to be available on every implementation of the Java platform" (https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html) so I don't think that is ever going to be a possibility. I am going to remove the try catch block.